### PR TITLE
Fix for Windows

### PR DESCRIPTION
**Pull request by beyondmetis**
Modified the gradleFetchArtifact routine to work with Windows systems. Tested on Windows 10 and on Mac.

### CODE REVIEW

**Comment by DavidBrainard** [comment]
This looks good.  I will merge when I'm back at my lab desktop later in the week, as I can then do a few quick tests that are a bit hard to do from where I am right now.

**Comment by DavidBrainard** [comment]
Some basic testing did not reveal any issues.  So I'm calling this good.

File: external/gradle/gradleFetchArtifact.m [hunk 1]
```diff
@@ -100,25 +100,23 @@
...
+            'Alternatively, you may download Git which bundles bash from https://git-scm.com/download/win. ' ...
             'Either way, make sure you add the path to bash to your windows environment path, then restart Matlab.' 
             ];
         error('FetchArtifact:BadStatus', msg, status, result)
     end
 end
 
-% Note that bash is implicit on Mac/Linux, but must be specified for
-% Windows
+% We must explicitly call bash when on Windows platforms. On Mac/Linux, the
```
Comment (floriancbeck) [inline external/gradle/gradleFetchArtifact.m]: ghjkhvk

**Comment by DavidBrainard** [comment]
I am unable to find this comment on github, nor able to understand it from this email in
isolation.

Can you expand?

Best,

David

**Comment by floriancbeck** [comment]
Hello David,

I am trying to execute the isetL3\tutorials\t_L3Object.m , which should fetch training data using the RemoteDataToolbox from a certain Maven repository. But I am failing to fetch the data from the repository due to the use of a Windows 10 system:

The call of the RemoteDataToolbox functions in the t_L3Object.m would be as follows:
> rd = RdtClient('scien');
> rd.crp('/L3/Farrell/D200/garden');
> s = rd.searchArtifacts('dsc_', 'type', 'pgm');
> 
> trainFile = 23;
> raw = im2double(rd.readArtifact(s(trainFile).artifactId, 'type', 'pgm'));
> rgb = im2double(rd.readArtifact(s(trainFile).artifactId, 'type', 'jpg'));

As I understood from the comments in line 116 f. in the gradleFetchArtifact.m, I replaced the string 'bash' with a explicit call of the Cygwin bash such that the first resulting command looks a follows:

C:\cygwin64\bin\bash.exe -c "DYLD_LIBRARY_PATH='' TERM=${TERM:-dumb} C:/Masterarbeit/Programming/RemoteDataToolbox-master/external/gradle/gradlew --daemon --refresh-dependencies -DREPOSITORY=http://52.32.77.154/repository/scien/ -DUSERNAME=guest -DPASSWORD= -DGROUP=L3.Farrell.D200.garden -DID=dsc_0791 -DVERSION=+ -DEXTENSION=pgm   -b C:/Masterarbeit/Programming/RemoteDataToolbox-master/external/gradle/fetch.gradle fetchIt"

As an output of the Matlab system(command) call I get the following:
status: 1
result: C:/Masterarbeit/Programming/RemoteDataToolbox-master/external/gradle/gradlew: line 137: C:/cygwin64/Masterarbeit/Programming/RemoteDataToolbox-master/external/gradle/fetch.gradle: No such file or directory

Therefore I have the question if I may have missed to configure something in Cygwin and/or Java. Somehow the path switching from line 112 on in the RemoteDataToolbox\external\gradle\gradlew while using Cygwin produces a path which does not exist. Did you encounter this issue while testing on a windows system while using Cygwin before?

Best regards,
Florian

**Comment by floriancbeck** [comment]
I have found a work around via using an environment variable I named _BASH_EXE_ for the Git bash.exe and changing the command in the `gradleFetchArtifact.m` than as follows:
`command = sprintf('%s "%s"', '"%BASH_EXE%" -c', command);`
This led to a successful call, but leads to another issue such that the download still fails:
```
Command : "%BASH_EXE%" -c "DYLD_LIBRARY_PATH='' TERM=${TERM:-dumb} C:/Masterarbeit/Programming/RemoteDataToolbox-master/external/gradle/gradlew --daemon --refresh-dependencies -DREPOSITORY=http://52.32.77.154/repository/scien/ -DUSERNAME=guest -DPASSWORD= -DGROUP=L3.Farrell.D200.garden -DID=dsc_0791 -DVERSION=+ -DEXTENSION=jpg   -b C:/Masterarbeit/Programming/RemoteDataToolbox-master/external/gradle/fetch.gradle fetchIt"
Status: 0
Result: Could not find artifactId <dsc_0791> of type <jpg> at remotePath <L3/Farrell/D200/garden>
```
Also with specifying the 'remotePath' as 'L3/Farrell/D200/garden/dsc_0791/1' in the `readArtifact.m` the fetch fails. Do you know if that is still Windows related or is there another reason for that?

Best regards.
Florian

**Comment by wandell** [comment]
Here is a link to the web-site:
http://52.32.77.154/#browse/L3

There does appear to be a directory with something like that string.

The rendering on the web-site looks like this:  Farrell.D200.garden<http://52.32.77.154/>  I believe, however, that Archiva shows nested directories with that ‘dot’ notation.

I would be glad to save you some effort by getting the data and providing it to you directly rather than going through the RDT methods.  I think we are likely to shift to Flywheel in the next six to 10 months anyway.  If it will save you a lot of hassle, I will try to create a download/zip file for you and provide you the images on the site.

Brian Wandell

**Comment by floriancbeck** [comment]
Thank you for the link to the web-site! For the start now I work then with downloading the files manually. 

Since I cannot foresee yet which data and further ISET toolboxes, which also access this and other repositories, I might need in the upcoming months, I would like to try to make the fetch work somehow. It seems like that the the the command is generated with the 'dot' notation for the nested directory `L3.Farrell.D200.garden` correctly.  The `searchArtifacts()` seems to work also fine and outputs the later used struct.  However, the `readArtifact()` fails maybe due to the versioning folder `1` of the files or some Windows specific problem. 

Or could it be a simply an authentication problem?

Best regards,
Florian

**Pull request merged**